File: api/src/main/java/com/api/app/entity/GoodsItem.java
package com.api.app.entity;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.Setter;
import org.apache.ibatis.type.Alias;

/**
 * @author system
 * @version 1.0
 * @since 2025-10-27
 */
@Alias("GoodsItem")
@Getter
@Setter
public class GoodsItem extends SystemEntity {
    private static final long serialVersionUID = 3345678901234567890L;

    @Schema(description = "상품번호")
    private String goodsNo;

    @Schema(description = "단품번호")
    private String itemNo;

    @Schema(description = "상품명")
    private String goodsName;

    @Schema(description = "단품금액")
    private Long itemPrice;

    @Schema(description = "재고수량")
    private Long stock;

    @Schema(description = "단품상태코드")
    private String goodsStatusCode;
}

File: api/src/main/kotlin/com/api/app/entity/GoodsItem.kt
package com.api.app.entity

import jakarta.persistence.*
import java.io.Serializable

@Embeddable
data class GoodsItemId(
    @Column(name = "GOODS_NO", length = 15) val goodsNo: String = "",
    @Column(name = "ITEM_NO", length = 3) val itemNo: String = ""
) : Serializable

@Entity
@Table(name = "GOODS_ITEM")
class GoodsItem : SystemEntity() {

    @EmbeddedId
    var id: GoodsItemId = GoodsItemId()

    @Column(name = "ITEM_NAME", nullable = false, length = 100)
    var itemName: String = ""

    @Column(name = "ITEM_PRICE", nullable = false, columnDefinition = "numeric")
    var itemPrice: Long = 0

    @Column(name = "STOCK", nullable = false, columnDefinition = "numeric")
    var stock: Long = 0

    @Column(name = "GOODS_STATUS_CODE", nullable = false, length = 3)
    var goodsStatusCode: String = ""

    // Convenience accessors
    val goodsNo: String get() = id.goodsNo
    val itemNo: String get() = id.itemNo
}
